### PR TITLE
RAB-1438: A job with multiple steps is able to resume from the paused step

### PR DESCRIPTION
**Pull request by ghoussard**
**Description (for Contributor and Core Developer)**

We made Job.php able to resume from the paused step execution when the job has multiple steps.
If there is a change in the job declaration (added or removed steps), we mark the job as failed.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Tool/Component/Batch/Job/Job.php [hunk 1]
```diff
@@ -345,4 +346,24 @@ private function deleteWorkingDirectory(string $directory)
...
+
+    private function getStepExecution(JobExecution $jobExecution, int $index): ?StepExecution
+    {
+        $stepExecution = $jobExecution->getStepExecutions()[$index] ?? null;
+
+        if(null === $stepExecution) {
+            return null;
+        }
+
+        if ($stepExecution->getStepName() !== $this->steps[$index]->getName()) {
```
Comment (ghoussard) [inline src/Akeneo/Tool/Component/Batch/Job/Job.php:363]: This check allow us to make the job fail if a steps has been added or removed from the job declaration during code upgrade. 
Feel free to propose a better exception message

Comment (ValentinMumble) [reply src/Akeneo/Tool/Component/Batch/Job/Job.php:363]: `Mismatched steps configuration`? Or something like `Steps do not match following deployment` 🤷

File: src/Akeneo/Tool/Bundle/BatchBundle/tests/Integration/Job/JobIntegration.php [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        $this->assertCount(2, $jobExecution->getStepExecutions());
+        $this->assertSame(BatchStatus::COMPLETED, $jobExecution->getStatus()->getValue());
+    }
+
+    public function test_it_handles_when_a_job_is_being_paused(): void
+    {
+        $steps = [
+            $this->createStep('step_1', BatchStatus::COMPLETED),
+            $this->createStep('step_2', BatchStatus::PAUSED),
+            $this->createStep('step_3'),
```
Comment (ValentinMumble) [inline src/Akeneo/Tool/Bundle/BatchBundle/tests/Integration/Job/JobIntegration.php:48]:
```suggestion
            $this->createStep('step_3', null),
```
Easier to read IMO

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Tool/Component/Batch/Job/Job.php (hunk 1 above)
Comment (MadeWilson) [reply src/Akeneo/Tool/Component/Batch/Job/Job.php:363]: Job step list has changed during pause, can't resume job started with old configuration.

**Review by MadeWilson — APPROVED** [review]

File: src/Akeneo/Tool/Component/Batch/Job/Job.php (hunk 1 above)
Comment (ghoussard) [reply src/Akeneo/Tool/Component/Batch/Job/Job.php:363]: Thanks!

**Pull request merged**